### PR TITLE
ci: support `explicit-function-return-type`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/components/input-password/input-password-options.ts [hunk 1]
```diff
@@ -17,9 +17,13 @@ export interface TuiInputPasswordOptions {
 export type InputPasswordOptions = TuiInputPasswordOptions;
 
 // TODO: remove in ivy compilation
-export const PASSWORD_ICON_HIDE = ({$implicit}: any) =>
+export const PASSWORD_ICON_HIDE = ({
+    $implicit,
```
Comment (nsbarsukov) [inline projects/kit/components/input-password/input-password-options.ts:21]: nit:

`TuiContextWithImplicit<TuiSize*>`

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-range/input-range.component.ts [hunk 2]
```diff
@@ -300,7 +300,9 @@ export class TuiInputRangeComponent
     }
 }
 
-export function tuiTextfieldAppearanceDirectiveFactory({nativeElement}: ElementRef) {
+export function tuiTextfieldAppearanceDirectiveFactory<T>({
+    nativeElement,
+}: ElementRef): T {
```
Comment (nsbarsukov) [inline projects/kit/components/input-range/input-range.component.ts:305]:
```suggestion
}: ElementRef): string {
```

File: projects/kit/internal/primitive-calendar-range/primitive-calendar-range.component.ts [hunk 3]
```diff
@@ -148,7 +148,7 @@ export class TuiPrimitiveCalendarRangeComponent implements OnInit {
         }
     }
 
-    private updatedViewedMonthSecond(month: TuiMonth) {
+    private updatedViewedMonthSecond(month: TuiMonth): TuiMonth | TuiDay {
```
Comment (nsbarsukov) [inline projects/kit/internal/primitive-calendar-range/primitive-calendar-range.component.ts:151]: i am not sure but I think `updatedViewedMonthSecond` and `updatedViewedMonthFirst` should return just `TuiMonth`

File: projects/demo/src/modules/animations/examples/fade-in/index.ts [hunk 4]
```diff
@@ -25,10 +25,7 @@ export class TuiFadeInExample {
     );
 
     @tuiPure
-    getAnimation(duration: number) {
-        return {
-            value: '',
-            params: {duration},
-        } as const;
+    getAnimation(duration: number): TuiDurationOptions {
+        return {value: '', params: {duration}} as const;
```
Comment (nsbarsukov) [inline projects/demo/src/modules/animations/examples/fade-in/index.ts:29]: do we need `as const` if we have return-type ?

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/animations/examples/fade-in/index.ts (hunk 4 above)
Comment (splincode) [reply projects/demo/src/modules/animations/examples/fade-in/index.ts]: fixed

File: projects/kit/components/input-range/input-range.component.ts (hunk 2 above)
Comment (splincode) [reply projects/kit/components/input-range/input-range.component.ts]: fixed

File: projects/kit/components/input-password/input-password-options.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/input-password/input-password-options.ts:21]: fixed

File: projects/addon-table/components/table/directives/sortable.directive.ts [hunk 5]
```diff
@@ -16,13 +16,13 @@ export class TuiSortableDirective<T> implements DoCheck {
         this.th.sorter = this.sorter;
     }
 
-    readonly sorter = () => 0;
+    readonly sorter = (): 0 => 0;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-table/components/table/directives/sortable.directive.ts]: fixed

**Pull request merged**